spring-integration-ftp/src/test/java/org/springframework/integration/ftp/outbound/FtpServerOutboundTests.java

```diff
@@ -182,6 +189,19 @@ public void testInt2866LocalDirectoryExpressionGET() {
 	}
 
 	@Test
+	public void testGetWithRemove() {
+		String dir = "ftpSource/";
+		this.getGw.setOption(Option.DELETE);
```

Comment:
Since you modify and existing `MH` and don't reinstate its behavior after this test, I think this is dangerous and may lead to an unexpected behavior, especially in those JVMs when method declaration order isn't supported.